StorageUtility/Linux/StorageUtility.cpp

```
@@ -17,251 +17,181 @@ limitations under the License.
+{
+    if(StorageAdapterProperty.AtaBus == true)
+        return true;
+    return false;
+}
 
-/*eErrorCode EnumerateDevices( sEnumerateDevicesCallback& Callback, const GUID* InterfaceClassGUID, eOnErrorBehavior OnErrorBehavior )
+bool InitStorageAdapterProperty( sStorageAdapterProperty& StorageAdapterProperty, udev_device* dev )
 {
-    HDEVINFO devInfoHandle =
-        SetupDiGetClassDevs
-        (
-        InterfaceClassGUID,
-        NULL,
-        NULL,
-        DIGCF_PRESENT | DIGCF_DEVICEINTERFACE
-        );
-    if ( INVALID_HANDLE_VALUE == devInfoHandle )
+    if( false == IsStorageDisk(dev))
     {
-        return( eErrorCode::Unknown );
+        return false;
     }
+    if(true == IsAtaDeviceBus(dev))
+    {
+        StorageAdapterProperty.AtaBus = true;
+        return true;
+    }
+    else
+    {
+        StorageAdapterProperty.AtaBus = false;
+        return false;
+    }
+    return true;
+}
 
-    DWORD error = 0;
-    DWORD memberIndex = 0;
-    while ( true == true )
+eErrorCode GetStorageDevicePaths( std::vector<String>& ipPaths, eOnErrorBehavior OnErrorBehavior )
+{
+    struct udev *udev;
+    struct udev_enumerate *enumerate;
+    struct udev_list_entry *devices, *dev_list_entry;
+    struct udev_device *dev;
+
+    /* Create the udev object */
+    udev = udev_new();
+    if (NULL == udev)
     {
-        SP_DEVICE_INTERFACE_DATA devInterfaceData;
-        devInterfaceData.cbSize = sizeof( devInterfaceData );
-        if ( SetupDiEnumDeviceInterfaces( devInfoHandle, NULL, InterfaceClassGUID, memberIndex++, &devInterfaceData ) == FALSE )
-        {
-            error = GetLastError();
-            if ( ERROR_NO_MORE_ITEMS == error )
-            {
-                break;
-            }
-            else
-            {
-                SetupDiDestroyDeviceInfoList( devInfoHandle );
-                return( eErrorCode::Unknown );
-            }
-        }
+        printf("ERROR: Can't create udev\n");
```

Do not print anything here
Please throw an exception instead